jdbc/src/test/java/org/apache/zeppelin/jdbc/JDBCInterpreterTest.java

```diff
@@ -658,4 +633,46 @@ public void testSplitSqlQueryWithComments() throws SQLException, IOException {
     assertEquals(InterpreterResult.Code.SUCCESS, interpreterResult.code());
     assertEquals(3, interpreterResult.message().size());
   }
+
+  @Test
+  public void testSelectQueryTypes() throws SQLException, IOException {
+    Properties properties = new Properties();
+    properties.setProperty("common.max_count", "1000");
+    properties.setProperty("common.max_retry", "3");
+    properties.setProperty("default.driver", "org.h2.Driver");
+    properties.setProperty("default.url", getJdbcConnection());
+    properties.setProperty("default.user", "");
+    properties.setProperty("default.password", "");
+    JDBCInterpreter t = new JDBCInterpreter(properties);
+    t.open();
+
+    String sqlQuery = "select * from test_table";
+
+    InterpreterResult interpreterResult = t.interpret(sqlQuery, interpreterContext);
+
+    assertEquals(InterpreterResult.Code.SUCCESS, interpreterResult.code());
+    assertEquals(Collections.nCopies(2, ColumnDef.TYPE.STRING),
+        interpreterResult.message().get(0).getColumnTypes());
```

Comment:
interpreter is not closed, it it better to put it in finally block

Reply:
Done